src/main/java/org/hisp/dhis/BaseDhis2.java

```diff
@@ -900,23 +905,36 @@ protected ObjectsResponse saveMetadataObjects( Objects objects )
      */
     protected ObjectResponse updateMetadataObject( String path, IdentifiableObject object )
     {
-        return updateObject( path, object, ObjectResponse.class );
+        return updateObject( path, Map.of( SKIP_SHARING_PARAM, true ), object, ObjectResponse.class );
     }
 
     /**
      * Updates an object using HTTP PUT.
      *
      * @param path the URL path relative to the API end point.
+     * @param queryParams the map of query parameters to include in the URL.
      * @param object the object to save.
      * @param type the class type for the response entity.
      * @param <T> class.
      * @return object holding information about the operation.
      */
-    protected <T extends BaseHttpResponse> T updateObject( String path, Object object, Class<T> type )
+    protected <T extends BaseHttpResponse> T updateObject( String path, Map<String, Object> queryParams, Object object,
+        Class<T> type )
     {
-        URI url = config.getResolvedUrl( path );
+        try
+        {
+            URI url = config.getResolvedUriBuilder().appendPath( path )
+                .addParameters( queryParams.entrySet().stream()
+                    .map( param -> new BasicNameValuePair( param.getKey(), param.getValue().toString() ) ).collect(
+                        Collectors.toList() ) )
+                .build();
 
-        return executeJsonPostPutRequest( new HttpPut( url ), object, type );
+            return executeJsonPostPutRequest( new HttpPut( url ), object, type );
+        }
+        catch ( URISyntaxException ex )
```

Comment:
Just curious, why do we need to catch invalid URI exceptions? I think the URI is controlled by the Java client, so we can ensure the URIs are correct?

Reply:
This is because `URIBuilder.build()` may throw an exception. A similar pattern plays out in `Dhis2Config.getResolvedUrl()`.